test/end2end_test.go

```diff
@@ -7244,32 +7244,38 @@ func (s) TestRPCWaitsForResolver(t *testing.T) {
 
 func (s) TestHTTPHeaderFrameErrorHandlingHTTPMode(t *testing.T) {
 	// Non-gRPC content-type fallback path.
-	for httpCode := range transport.HTTPStatusConvTab {
-		doHTTPHeaderTest(t, transport.HTTPStatusConvTab[int(httpCode)], []string{
+	for httpCode, statusCode := range transport.HTTPStatusConvTab {
+		if err := doHTTPHeaderTest(t, statusCode, []string{
 			":status", fmt.Sprintf("%d", httpCode),
 			"content-type", "text/html", // non-gRPC content type to switch to HTTP mode.
 			"grpc-status", "1", // Make up a gRPC status error
 			"grpc-status-details-bin", "???", // Make up a gRPC field parsing error
-		})
+		}); err != nil {
+			t.Fatal(err)
```

Comment:
Do these need to be t.Fatal() instead of t.Error()? (here and elsewhere in this test)

Reply:
This is obsolete after reverting the refactoring.